runtime/src/vm/memory.rs

```diff
@@ -111,67 +106,171 @@ impl HandlerMemory {
+          self.mem[reg] = arr2.mem[arr_addr2 as usize];
+        } else {
+          // Make a pointer to this data
+          if arr.either_mem[arr_addr2 as usize] == -1 {
+            ptr = arr.mem.as_ptr().add(arr_addr2 as usize) as i64;
+            self.registers_ish[reg] = FX_PTR_ADDR;
+          } else {
+            ptr = arr.fractal_mem.as_ptr().add(arr.either_mem[arr_addr2 as usize] as usize) as i64;
+            self.registers_ish[reg] = HM_PTR_ADDR;
+          }
+          self.mem[reg] = ptr;
+        }
+      }
+    }
+  }
+
+  pub fn store_reg(self: &mut HandlerMemory, reg_addr: i64, other_addr: i64) {
+    let ptr: *const HandlerMemory = self.get_fractal(other_addr);
+    self.registers_ish[reg_addr as usize] = HM_PTR_ADDR;
+    self.mem[reg_addr as usize] = ptr as i64;
+  }
+
+  /// optionally create a registerish within a nested Result-like HandlerMemory, or leave an error
+  /// message. This is here instead of in the opcodes because of the special access it needs to
+  /// private fields
+  pub fn res_from(self: &mut HandlerMemory, arr_addr: i64, arr_idx_arr: i64, outer_addr: i64) {
+    self.new_fractal(outer_addr);
+    self.push_fractal_fixed(outer_addr, 1i64);
+    let inner_addr = self.read_fixed(arr_idx_arr);
+    let arr = self.get_fractal(arr_addr);
+    if inner_addr >= 0 && arr.len() as i64 > inner_addr {
+      // Valid result
+      unsafe {
+        let ptr;
+        let ptr_type;
+        if arr.either_mem[inner_addr as usize] == -1 {
+          ptr = arr.mem.as_ptr().add(inner_addr as usize) as i64;
+          ptr_type = FX_PTR_ADDR;
+        } else {
+          ptr = arr.fractal_mem.as_ptr().add(arr.either_mem[inner_addr as usize] as usize) as i64;
+          ptr_type = HM_PTR_ADDR;
+        }
+        self.push_fractal_fixed(outer_addr, 0i64);
+        let res_arr = self.get_mut_fractal(outer_addr);
+        res_arr.registers_ish[1] = ptr_type;
+        res_arr.mem[1] = ptr;
+      }
+    } else {
+      // Out-of-bounds access
+      let error_string = "out-of-bounds access";
+      let mut out = vec![error_string.len() as i64];
+      let mut out_str_bytes = error_string.as_bytes().to_vec();
+      loop {
+        if out_str_bytes.len() % 8 != 0 {
+          out_str_bytes.push(0);
+        } else {
+          break
+        }
+      }
+      let mut i = 0;
+      loop {
+        if i < out_str_bytes.len() {
+          let str_slice = &out_str_bytes[i..i+8];
+          out.push(i64::from_ne_bytes(str_slice.try_into().unwrap()));
+          i = i + 8;
+        } else {
+          break
+        }
       }
-      self.mem[reg] = ptr;
+      let arr = self.get_mut_fractal(outer_addr);
+      arr.mem[0] = 0;
+      self.push_nested_fractal_mem(outer_addr, out);
     }
   }
 
   /// The address provided can be a directly nested fractal or a registerish address that points to
   /// a fractal. Either returns a reference to a HandlerMemory
   pub fn get_fractal(self: &HandlerMemory, addr: i64) -> &HandlerMemory {
```

Comment:
can we collapse `read_fractal` and `get_fractal` what is the difference?

Reply:
`get_fractal` returns a reference to the actual HandlerMemory, `read_fractal` returns a copy of it. Sometimes one is useful versus the other. I do agree that I want to trim this API again, but I intentionally wanted to keep that to a "cleanup" PR as mentioned above where you know all of the changes are supposed to produce identical behavior.